### PR TITLE
🌱 Log error when calling a Runtime Extension gets an error that is ignored because of failure policy

### PR DESCRIPTION
**Pull request by tsuzu**
**What this PR does / why we need it**:
* Log an error when Lifecycle Hook Runtime Extension fails while we set `Ignore` as Failure Policy. 
    * I want the error to be on logs even if ignored for investigation.

```
E0808 14:00:07.430045       1 client.go:349] "ignoring error calling extension handler because of FailurePolicy \"Ignore\"" err="http call failed: Post \"https://test-extension-webhook-service.test-extension-system.svc:443/hooks.runtime.cluster.x-k8s.io/v1alpha1/beforeclustercreate/before-cluster-create?timeout=10s\": dial tcp 10.96.93.238:443: connect: connection refused" controller="topology/cluster" controllerGroup="cluster.x-k8s.io" controllerKind="Cluster" Cluster="default/capi-quickstart" namespace="default" name="capi-quickstart" reconcileID="8dc4c6bd-f70b-4179-bc80-0df5afcfbf6c" hook="BeforeClusterCreate" extensionHandler="before-cluster-create.test-runtime-sdk-extensionconfig" hook="BeforeClusterCreate"
```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

### CODE REVIEW

File: internal/runtime/client/client.go [hunk 1]
```diff
@@ -346,7 +346,7 @@ func (c *client) CallExtension(ctx context.Context, hook runtimecatalog.Hook, fo
 		ignore := *registration.FailurePolicy == runtimev1.FailurePolicyIgnore
 		if _, ok := err.(errCallingExtensionHandler); ok && ignore {
 			// Update the response to a default success response and return.
-			log.Info(fmt.Sprintf("ignoring error calling extension handler because of FailurePolicy %q", *registration.FailurePolicy))
+			log.Info(fmt.Sprintf("ignoring error calling extension handler because of FailurePolicy %q", *registration.FailurePolicy), "error", err)
```
Comment (sbueringer) [inline internal/runtime/client/client.go:349]: please use log.Error instead

Comment (tsuzu) [reply internal/runtime/client/client.go]: OK!
Fixed at https://github.com/kubernetes-sigs/cluster-api/pull/11028/commits/6cc06161c2c631c363c882abfa89b1464a217b22

**Comment by sbueringer** [comment]
Thx!

/lgtm
/approve

**Pull request merged**